test/renderUtils.tsx

```diff
@@ -1,20 +1,25 @@
-import { render } from '@testing-library/react';
+import { render, RenderOptions } from '@testing-library/react';
 import App from 'App';
 import MapiAuthProvider from 'Auth/MAPI/MapiAuthProvider';
 import { ConnectedRouter } from 'connected-react-router';
 import { createMemoryHistory } from 'history';
+import { History } from 'history';
+import { IOAuth2Provider } from 'lib/OAuth2/OAuth2';
 import TestOAuth2 from 'lib/OAuth2/TestOAuth2';
 import React from 'react';
 import { Provider } from 'react-redux';
 import { MainRoutes } from 'shared/constants/routes';
 import configureStore from 'stores/configureStore';
 import { LoggedInUserTypes } from 'stores/main/types';
+import { IState } from 'stores/state';
 import theme from 'styles/theme';
 import ThemeProvider from 'styles/ThemeProvider';
 import { SWRConfig } from 'swr';
 import { FlashMessagesController } from 'UI/Util/FlashMessages/FlashMessagesController';
 import FlashMessagesProvider from 'UI/Util/FlashMessages/FlashMessagesProvider';
 
+type PropsOf<C> = C extends React.ComponentType<infer P> ? P : never;
```

Comment:
This infers the props from a component. It's used with all rendering utilities to be able to provide typechecking for props in tests